### ISSUE TITLE
Features are only deprecated for a month before being removed

### ISSUE THREAD

**Issue by medavox** [jmockit/jmockit1#494]
Please provide the following information:

* Version of JMockit that was used: 
Version 1.37
* Description of the problem:

I used this library last year 
(version 1.30, so presumably between 18th December 2016 and 26th March 2017).

Back then, the `Deencapsulation` class provided an `invoke` method.
I need this for my use case.

I checked your release history, and that feature was only deprecated in 24th September 2017, before being removed 1 month and 5 days later in version 1.36 (29th October 2017).

*1 month is not a long enough deprecation period for API users like me to migrate their code.*

I'm **not** asking you to reverse your change to the API, I'm just asking for more warning, the next time there are breaking changes.

I know I can use an older version of the library, but using unsupported software is never a good idea.
Plus, if I needed a feature from a newer version, I would be stuck. Not to mention security implications.

My company needs me to write tests which continue to function for a long period of time (eg years), while also writing newer tests which use the latest & best features currently available.

I understand that you are donating your time to write an extremely useful and helpful library (and for that: thank you!), and that API changes are entirely your prerogative.

But I ask you to consider the users of that API, like myself, who need the API to remain stable enough for real-life business use cases in production software. This library is version 1, which led me to assume a certain level of API stability.

TL;DR My request is: Please give **at least** 3 months notice before making breaking API changes in future. Preferably 6 months.

I know you have no obligation to do anything.

The worst thing that can happen for me is that I have to fork and maintain an older version. But this significantly increases my own workload. And if every library I use did that, then I would be buried under a mountain of stable forks of other people's code.

Thank you again for a well-written and useful piece of software.
Peace

**Comment by rliesenfeld** [comment]
Ok, 3 months sounds good. At this point, there aren't going to be any API changes in JMockit 1.x anymore, except perhaps for the eventual removal of the `MockUp#getMockInstance()` method by the end of february or later.

Next year will hopefully see the initial version of JMockit 2, which will have a smaller API and more stability (given now that many API design lessons have been learned, and that features are not going to be added just because it can be done, as often happened in the past decade).
